packages/sdk-react/src/components/receivables/InvoiceDetails/ExistingInvoiceDetails/components/InvoiceCancelModal.tsx

```diff
@@ -0,0 +1,95 @@
+import { useRootElements } from '@/core/context/RootElementsProvider';
+import {
+  useCancelReceivableById,
+  useReceivableById,
+} from '@/core/queries/useReceivables';
+import { t } from '@lingui/macro';
+import { useLingui } from '@lingui/react';
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Divider,
+  Skeleton,
+} from '@mui/material';
+
+interface InvoiceCancelModalProps {
+  invoiceId: string;
+  open: boolean;
+  onClose: () => void;
+}
+
+export const InvoiceCancelModal = ({
+  invoiceId,
+  open,
+  onClose,
+}: InvoiceCancelModalProps) => {
+  const { i18n } = useLingui();
+  const { root } = useRootElements();
+  const { data: receivable, isLoading: isReceivableLoading } =
+    useReceivableById(invoiceId);
+  const cancelReceivableMutation = useCancelReceivableById(invoiceId);
+
+  const creditNoteIds =
+    receivable?.type === 'invoice'
+      ? receivable.related_documents.credit_note_ids
+      : undefined;
+
+  const hasNoCreditNotes = !creditNoteIds?.length;
+
+  const modalTitle = !receivable ? (
+    <Skeleton />
+  ) : receivable.document_id ? (
+    t(i18n)`Cancel receivable "${receivable.document_id}"?`
+  ) : (
+    t(i18n)`Cancel receivable?`
+  );
+
+  const modalDescription = hasNoCreditNotes
+    ? t(
+        i18n
+      )`The Credit note to this invoice ${receivable?.document_id} was created earlier. Following that you can’t cancel invoice.`
+    : t(i18n)`This action can't be undone.`;
+
+  return (
+    <Dialog
+      open={open && Boolean(invoiceId)}
+      container={root}
+      onClose={onClose}
+      aria-label={t(i18n)`Invoice cancel confirmation`}
+      maxWidth="sm"
+      fullWidth
+    >
+      <DialogTitle variant="h3">
+        {hasNoCreditNotes
+          ? modalTitle
+          : t(i18n)`Unfortunately, you can't cancel`}
+      </DialogTitle>
+      <DialogContent>{modalDescription}</DialogContent>
+      <Divider />
+      <DialogActions>
+        <Button variant="outlined" onClick={onClose} color="inherit">
+          {t(i18n)`Close`}
+        </Button>
+        {hasNoCreditNotes && (
+          <Button
+            variant="outlined"
+            color="error"
+            disabled={cancelReceivableMutation.isPending || isReceivableLoading}
+            onClick={(event) => {
+              event.preventDefault();
+              cancelReceivableMutation.mutate(undefined, {
```

Comment:
We need to add tests for the `InvoiceCancelModel` to ensure that the invoice cancellation process works as expected.

## Required Tests
1. **Invoice Cancellation Button**
   - Tests should verify that the cancellation button is correctly displayed based on whether the invoice meets the criteria.
   
2. **Invoice Cancellation Modal**
   - Tests should confirm that clicking the cancellation button correctly triggers the modal.
   - The modal should then successfully cancel the invoice.

### Scenarios to Cover
- **Scenario 1:** The invoice meets the criteria for showing the cancellation button.
- **Scenario 2:** The invoice does not meet the criteria for showing the cancellation button.